File: 1000/Rudolf_and_121.py
"""
Author: passa
Task #: 1941B
Task Name: Rudolf and 121
Difficulty: 1000
Language: Python
"""

def solution(lst):
    for i in range(1, len(lst) - 1):
        if lst[i-1] == 0:
            continue
        if lst[i-1] * 2 > lst[i]:
            return "NO"
        
        lst[i] -= lst[i-1] * 2
        lst[i+1] -= lst[i-1]

    if lst[len(lst) - 1] == 0 and lst[len(lst) - 2] == 0:
        return "YES"
    else: return "NO"

def read_input():
    n = int(input())

    lst = [int(a) for a in input().split()]

    return lst

### main ###

t = int(input())

for i in range(t):
    lst = read_input()
    print(solution(lst))

File: 1000/Rudolf_and_121.cpp
/*
Author: passa
Task #: 1941B
Task Name: Rudolf and 121
Difficulty: 1000
Language: C++ 
*/
#include <bits/stdc++.h>
using namespace std;

int t,n,a;
vector<int> A;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    cin >> t;

    while (t--) {
        cin >> n;
        for (int i = 0 ; i < n ; i++) {
            cin >> a;
            A.push_back(a);
        }
        
        for (int i = 1 ; i < n - 1 ; i++) {
            if (A[i - 1] == 0) {
                continue;
            }
            if (A[i - 1] * 2 > A[i]) {
                cout << "NO" << endl;
                break;
            }

            A[i] -= A[i - 1] * 2;
            A[i + 1] -= A[i - 1];
        }

        if (A[n - 1] == 0 && A[n - 2] == 0) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }

        A.clear();
    }
    
    return 0;
}